File: vjezbe_3/particle.py
import math as m 
import matplotlib.pyplot as plt

class particle:
    def __init__(self,v0,kut,x0,y0):
        self.v0 = v0
        kut = (kut/180)*m.pi
        self.kut = kut
        self.x0 = x0
        self.y0 = y0
        self.vx = self.v0*m.cos(self.kut)
        self.vy = self.v0*m.sin(self.kut)
        self.pomak_x = []
        self.pomak_y = []

    def reset(self):
        del self.v0
        del self.kut
        del self.x0
        del self.y0
        del self.vx
        del self.vy
        del self.pomak_x
        del self.pomak_y

    def __move(self,dt):
        self.vy = self.vy - 9.81*dt
        self.x0 = self.x0 + self.vx * dt
        self.pomak_x.append(self.x0)
        self.y0 = self.y0 + self.vy * dt
        self.pomak_y.append(self.y0)

    def range(self,dt):
        pocetni_polozaj = self.x0
        while True:
            self.__move(dt)
            if self.y0 <= 0:
                break
        domet = self.pomak_x[-1] - pocetni_polozaj
        return domet

    def plot_trajectory(self):
        x_cord = [self.pomak_x]
        y_cord = [self.pomak_y]
        plt.scatter(x_cord,y_cord,s=1)
        plt.xlabel('domet(m)')
        plt.ylabel('visina(m)')
        plt.show()

File: vjezbe_7/particle.cpp
#include <particle.h>
#include <math.h>
using namespace std;

void Particle::pomak(){
        vy = vy - 9.81*dt;
        x = x + vx * dt;
        y = y + vy * dt;
        };

void Particle::restart(){
        vx = v0 * cos(theta);
        vy = v0 * sin(theta);
        x = x0;
        y = y0;
        };

Particle::Particle (double brzina, double kut, double polozaj_x, double polozaj_y ,double vrijeme){
        v0 = brzina;
        theta = (kut/180)*3.14159;
        x0 = polozaj_x;
        y0 = polozaj_y;
        vx = v0 * cos(theta);
        vy = v0 * sin(theta);
        x = x0;
        y = y0;
        dt = vrijeme;
        };

double Particle::range(){
        while (y >= 0){
            pomak();
            }
        double d = x - x0;
        restart();
        return d;
        };

double Particle::time(){
        double t = 0.0;
        while (y >= 0){
            pomak();
            t = t + dt;
            }
        restart();
        return t;
        };


